### DISCUSSION TITLE
fetch error during demo-image-sato build, kirkstone branch

### DISCUSSION THREAD

**Discussion by hutcruchi** [OE4T/meta-tegra#1322] [Q&A] [answered]
Hi,
I'm trying to build demo-image-sato and getting a fetch error during bitbake execution (I followed the instructions in the tegra-demo-distro README)

terminal.log:
[terminal.log](https://github.com/OE4T/meta-tegra/files/12334341/terminal.log)

error.log:
[log.do_fetch.907290.log](https://github.com/OE4T/meta-tegra/files/12334342/log.do_fetch.907290.log)

Thanks

**Comment by madisongh** [comment]
From the do_fetch log:
```
Resolving deltas:  80% (5377655/6694800)
error: index-pack died of signal 9
fatal: index-pack failed
```

Signal 9 is SIGKILL, so something killed one of the git subprocesses.

**Reply by hutcruchi** [reply]
I don't know who and why but I noticed my Ubuntu hanged sometimes. Maybe the OS killed one of the processes? 

1. Can I limit the resources used by bitbake build in order to prevent system hang? (build slower)
2. Can I build again without building everything from start, just what is missing? (to build faster)

Thanks

**Reply by madisongh** [reply, marked as answer]
There are some variables you can set to limit parallelism if your build host is short on resources. [This section](https://docs.yoctoproject.org/4.0.11/singleindex.html#speeding-up-a-build) of the Yocto Project documentation mentions them. By default, the bitbake sets the parallelism variables based on the number CPU cores/threads available.   It's generally recommended to have at least 1GiB RAM available per core/thread, IIRC.

The build system already has built-in mechanisms for avoiding unneeded rebuilds of packages.  I'd really recommend visiting [the Yocto Project](https://www.yoctoproject.org/) web site for more information about the build system, host requirements, etc. The on-line documentation is well-maintained, and you'll also find links there to books, presentations, and training.

**Reply by hutcruchi** [reply]
Thank you for your answer. 
I verified with dmesg it was indeed a memory issue , hence the SIGKILL.
I used the variables you mentioned. 
Regarding reading all the learnning sources -  I'll do it. I try to "quick start" with the tegra demo and than I'll dive deeper. 
If you have any recommendation regarding specific practical tutorial it will be much helpful (there is a lot information out there).

**Reply by Deepak-Beniya** [reply]
hlo @hutcruchi how did you solved the problem ? I am also fasing the same issue like you got. 
I am new to this Yocto project could you tell the steps for building the images for orin devkit board ?
where to find the .sdcard image ? 
That would be great help to me .


Thanks and regards
Deepak Kumar Beniya

**Reply by dwalkes** [reply]
> I am new to this Yocto project could you tell the steps for building the images for orin devkit board ?

See https://github.com/OE4T/tegra-demo-distro for step by step build instructions with a demo distribution and https://github.com/OE4T/meta-tegra/wiki/Flashing-the-Jetson-Dev-Kit for flashing instructions.

**Reply by Deepak-Beniya** [reply]
[error_1.txt](https://github.com/OE4T/meta-tegra/files/14232512/error_1.txt)
[error_2.txt](https://github.com/OE4T/meta-tegra/files/14232513/error_2.txt)

[log.do_fetch.2552113.txt](https://github.com/OE4T/meta-tegra/files/14232728/log.do_fetch.2552113.txt)
[log.do_fetch.2552205.txt](https://github.com/OE4T/meta-tegra/files/14232729/log.do_fetch.2552205.txt)



Summary: 3 tasks failed:
  /home/deepak/fiverr/client2/yocto-tegra/tegra-demo-distro/layers/meta-tegra/recipes-bsp/uefi/standalone-mm-optee-tegra_35.4.1.bb:do_fetch
  /home/deepak/fiverr/client2/yocto-tegra/tegra-demo-distro/layers/meta-tegra/recipes-kernel/linux/linux-tegra_5.10.bb:do_fetch
  /home/deepak/fiverr/client2/yocto-tegra/tegra-demo-distro/layers/meta-tegra/recipes-bsp/uefi/edk2-firmware-tegra_35.4.1.bb:do_fetch
Summary: There were 9 WARNING messages.
Summary: There were 6 ERROR messages, returning a non-zero exit code.


Check the error logs I have  attached .
It would be greatfull if you could help out.

**Reply by dwalkes** [reply]
Looks like network issues:
```
Receiving objects:   3% (11681/358174), 4.68 MiB | 56.00 KiB/s
Receiving objects:   3% (11699/358174), 4.68 MiB | 50.00 KiB/s
Receiving objects:   3% (11990/358174), 4.74 MiB | 57.00 KiB/s
error: RPC failed; curl 92 HTTP/2 stream 0 was not closed cleanly: CANCEL (err 8)
fatal: the remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
WARNING: Failed to fetch URL gitsm://github.com/NVIDIA/edk2.git;protocol=https;branch=r35.4.1-edk2-stable202208-updates;name=edk2;destsuffix=edk2-tegra/edk2;nobranch=1, attempting MIRRORS if available
```
Are you behind a VPN or unstable internet connection perhaps?

**Reply by Deepak-Beniya** [reply]
No, I am not using VPN and the network is stable. 
I went through the document where will be the output image ?
Do I abel to get .sdcard image ?

**Reply by dwalkes** [reply]
Can you clone these repositories successfully on the same host outside your Yocto build?

**Reply by Deepak-Beniya** [reply]
yes 
I have followed this readme https://github.com/OE4T/tegra-demo-distro 
Is there any video or proper document to build the images for orin nano board. beacause i went through the tegra-demo-distro readme but i am facing the error.

I want to build all the images .

IS there any possibility that I can share all the images(primary,secondary,Zimage,dtb , rootfs) to my friend  through one single file ?
Directory for all the images ?
